exporters/webhook/plugins/pelorus_handler.py

```diff
@@ -0,0 +1,168 @@
+        return payload_model
+
+    def _pelorus_deploytime(payload) -> DeployTimePelorusPayload:
+        payload_model = DeployTimePelorusPayload(**payload)
+        return payload_model
+
+    # Mapping between event_type given by the
+    # X-Pelorus-Event that is stored in the PelorusDeliveryHeaders
+    # and functions for its' relevant pydantic payload models
+    #
+    # For 'ping' X-Pelorus_event a pong classmethod that raises
+    # HTTPException to send 'pong' response is used.
+    handler_functions = {
+        PelorusMetricSpec.PING: PelorusWebhookResponse.pong,
+        PelorusMetricSpec.COMMIT_TIME: _pelorus_committime,
+        PelorusMetricSpec.FAILURE: _pelorus_failure,
+        PelorusMetricSpec.DEPLOY_TIME: _pelorus_deploytime,
+    }
+
+    @override
+    async def _handshake(self, headers: Headers) -> Awaitable[bool]:
+        """
+        Initial handshake implementation called by the plugin's base handler
+        method. The headers must match the PelorusDeliveryHeaders model to
+        be recognized by pydantic as valid headers, otherwise exception
+        is raised to inform user agent about improper headers immediately.
+
+        Returns:
+            bool: True when the handshake based on the headers were success
+
+        Raises:
+            HTTPException: headers were properly validated by pydantic
+        """
+        try:
+            self.payload_headers = parse_obj_as(PelorusDeliveryHeaders, headers)
+            return issubclass(type(self.payload_headers), PelorusDeliveryHeaders)
+        except ValidationError as ex:
+            logging.error(headers)
+            logging.error(ex)
+            raise HTTPException(
+                status_code=http.HTTPStatus.BAD_REQUEST,
+                detail="Improper headers.",
+            )
+
+    @override
+    async def _receive_pelorus_payload(
+        self, json_payload_data: Any
+    ) -> Awaitable[PelorusMetric]:
+        """
+        Receive payload from the json_payload_data and converts it to the
+        proper PelorusMetric by using mapping from the handler_functions.
+
+
+        Returns:
+            Awaitable[PelorusMetric]: with the proper Pelorus payload data.
+
+        Raises:
+            HTTPException: If the json_payload was not in a format required
+                           by the handler function requested for that payload
+                           in the header's 'X-Pelorus_event' event_type.
+        """
+        if self.payload_headers and self.payload_headers.event_type:
+            try:
+                data = self.handler_functions.get(
+                    self.payload_headers.event_type, lambda payload: None
+                )(json_payload_data)
+                return PelorusMetric(
+                    metric_spec=self.payload_headers.event_type, metric_data=data
+                )
+            except ValidationError as ex:
+                logging.error(self.payload_headers)
```

Comment:
uvicorn has its own logger, should we use the same?
![Screenshot from 2023-03-01 17-44-57](https://user-images.githubusercontent.com/66965232/222261768-ca2ff66d-ecea-4169-b229-3ef2eecafd4b.png)

Reply:
We could if we *really* wanted, but that comes at a cost of handling many `uvicorn`, `gunicorn` loggers and passing our streams to those. I don't think it's worth here.

See this thread: https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker/issues/19

Also I want to use setup_logging, so the logging from other exporters are same, this isn't really expected fromat by the uvicorn handler. I think this isn't important enough to be resolved now, we can always address this issue later if needed, by reworking logging mechanism for all the exporters and providing different handlers for the terminal and uvicorn.

Reply:
Another thing is separation between http events and internal application validations.

Here the log should be:

INFO for http event (we have OK message saying 422 error occured), so we sent OK back to the requestor with the "{"detail":"Invalid payload."}" message, and we log this as INFO, because the response was OK.

ERROR from validations, to know in logs *why* the data was invalid, but this is only information for user to see why the payload was invalid, we should not tell the requester *why* we should just say *bye bye invalid payload*. So the ERROR is not really uvicorn error, uvicorn was OK.